scrape/scrape.go

```diff
@@ -133,8 +133,75 @@ var (
 			Help: "How many times a scrape cache was flushed due to getting big while scrapes are failing.",
 		},
 	)
+	targetMetadataCache = newMetadataMetricsCollector()
 )
 
+// MetadataMetricsCollector is a Custom Collector for the metadata cache metrics.
+type MetadataMetricsCollector struct {
+	CacheEntries *prometheus.Desc
+	CacheBytes   *prometheus.Desc
+
+	scrapeCaches map[*Target]*scrapeCache
```

Comment:
Don't maintain an additional structure, that's going to be fragile. Instead walk what we already have.

Reply:
As Julien mentioned above, the alternative would be walking over the loops to get the cache. Is that what you have in mind or is it something different? 

We'd still need to hold an additional structure unless I'm missing something.

Reply:
That's what I have in mind.

I don't see why you'd need anything beyond a reference to the target manager.